core/predicate_check_test.go

```diff
@@ -80,23 +83,53 @@ func TestCheckPredicate(t *testing.T) {
 		},
 		"proposer predicate, no access list passes": {
 			address:            common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"),
+			gas:                53000,
 			proposerPredicater: &mockProposerPredicater{predicateFunc: func(*precompileconfig.ProposerPredicateContext, []byte) error { return nil }},
 			expectedErr:        nil,
 		},
 		"predicate, no access list passes": {
 			address:     common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"),
+			gas:         53000,
 			predicater:  &mockPredicater{predicateFunc: func(*precompileconfig.PrecompilePredicateContext, []byte) error { return nil }},
 			expectedErr: nil,
 		},
 		"predicate with valid access list passes": {
 			address: common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"),
-			predicater: &mockPredicater{predicateFunc: func(_ *precompileconfig.PrecompilePredicateContext, b []byte) error {
-				if bytes.Equal(b, common.Hash{1}.Bytes()) {
-					return nil
-				} else {
-					return fmt.Errorf("unexpected bytes: 0x%x", b)
-				}
-			}},
+			gas:     53000,
+			predicater: &mockPredicater{
+				predicateFunc: func(_ *precompileconfig.PrecompilePredicateContext, b []byte) error {
+					if bytes.Equal(b, common.Hash{1}.Bytes()) {
+						return nil
+					} else {
```

Comment:
style nit: we can drop one branch here in favor of early return (+ other occurrences in this file)

Reply:
Yup good catch